lua/neo-tree/ui/renderer.lua

```diff
@@ -17,6 +17,27 @@ local default_popup_size = { width = 60, height = "80%" }
 local floating_windows = {}
 local draw, create_window, create_tree, render_tree
 
+-- clean up neotree buffers created by :mksession
+local cleaned_up = false
+local clean_neotree_buffers = function()
+  if cleaned_up then
+    return
+  end
+  local clean_buffer_with_prefix = function(prefix)
+    for bufnr = 1, vim.fn.bufnr("$") do
+      local bufname = vim.fn.bufname(bufnr)
+      if string.sub(bufname, 1, #prefix) == prefix then
+        vim.api.nvim_buf_delete(bufnr, { force = true })
+      end
+    end
+  end
+  local sources = { "filesystem", "buffers", "git_status" }
```

Comment:
There could be more sources than this. It's best to just iterate over all buffers and use the pattern in the previous comment instead of looking for these specific source names.